test/test_ops.py

```diff
@@ -160,6 +161,34 @@ def test_numpy_ref(self, device, dtype, op):
         finally:
             torch.set_default_dtype(cur_default)
 
+    # Tests that the cpu and gpu results are consistent
+    @onlyCUDA
+    @suppress_warnings
+    @ops(_ops_and_refs_with_no_numpy_ref, dtypes=OpDTypes.any_one)
+    def test_compare_cpu(self, device, dtype, op):
+
+        if dtype not in op.dtypes:
+            raise unittest.SkipTest("This test requires both cuda and cpu support")
+        if dtype not in op.dtypesIfCUDA:
+            raise unittest.SkipTest("This test requires both cuda and cpu support")
```

Comment:
`any_one` will give the first dtype available for, in this case, CUDA. That does not mean that there isn't a dtype in the intersection of CPU and CUDA dtypes.

Reply:
A potential workaround: 

```python
    @ops(_ops_and_refs_with_no_numpy_ref, dtypes=OpDTypes.none)
    def test_compare_cpu(self, device, dtype, op):
        dtype_choices = op.dtypes.intersection(op.dtypesIfCUDA)
        if len(dtype_choices) == 0: 
            raise unittest.SkipTest("This test requires both cuda and cpu support")
        dtype = next(iter(dtype_choices)) # or dtype_choices.pop() as you like
```

Basically you would ensure the test is only instantiated for single `dtype` value (namely `dtype=None`) and then find an appropriate `dtype` for the test if one exists internally.

Reply:
Isn't that what `OpDtypes.any_one` does?

I think this check is redundant with the requested `dtypes`.

Reply:
After talking with amjames, I updated the comment. any_one chooses `float32` if available, please let me know if you want me to update with the proposed solution

Reply:
After trying this out, it seems like `toleranceOverride` does not work with `dtypes=OpDTypes.none`

Reply:
Yeah, that's fair. I bumped into a similar issue sometime ago and I ended up having to fix `any_one`.

Reply:
That's true that we don't have good support for controlling the tolerances of tests whose dtype is determined by `any_one`, and you're right we have no mechanism for controlling it using decorators when the test has no dtype. We could add an 'any' or 'all' option to the toleranceOverride decorator to address this

Reply:
resolved in another PR on the stack!